sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -376,6 +432,9 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
 
   static const char *PiTrace = std::getenv("SYCL_PI_TRACE");
   static const int PiTraceValue = PiTrace ? std::stoi(PiTrace) : 0;
+  PiESimdSurfaceMap = new std::unordered_map<unsigned int, _pi_mem *>;
```

Comment:
what if `piPlatformsGet` get invoked multiple times?
why not follow approach used in level_zero.cpp (which was apparently as blueprint for this) and initialize PiESimdSurfaceMap and other globals at the point of declaration?

Same applies to `CmBuf` creation below, which I overlooked last time. Note that in `level_zero.cpp` all initialization done within `piPlatformsGet` is guarded with checks.